recipes/readerwriterqueue/all/conanfile.py

```diff
@@ -11,27 +16,28 @@ class ReaderWriterQueue(ConanFile):
     license = "BSD-2-Clause"
     no_copy_source = True
     settings = "os"
-    
-    @property
-    def _source_subfolder(self):
-        return "sources_subfolder"
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("readerwriterqueue-{}".format(self.version),
-                  self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def package(self):
-        self.copy("atomicops.h", src=self._source_subfolder, dst=os.path.join("include", "readerwriterqueue"))
-        self.copy("readerwriterqueue.h", src=self._source_subfolder, dst=os.path.join("include", "readerwriterqueue"))
-        if tools.Version(self.version) >= "1.0.5":
-            self.copy("readerwritercircularbuffer.h", src=self._source_subfolder, dst=os.path.join("include", "readerwriterqueue"))
+    def build(self):
+        pass
 
-        self.copy("LICENSE.md", src=self._source_subfolder, dst="licenses")
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "atomicops.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include", "readerwriterqueue"))
+        copy(self, "readerwriterqueue.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include", "readerwriterqueue"))
+        if Version(self.version) >= "1.0.5":
+            copy(self, "readerwritercircularbuffer.h", src=self.source_folder, dst=os.path.join("include", "readerwriterqueue"))
```

Comment:
dst is incorrect here, related to https://github.com/conan-io/conan-center-index/issues/16651?

Reply:
Thanks! I saw it was fixed in https://github.com/conan-io/conan-center-index/pull/16672.